src/dynamic_boundary_conditions/hyetograph.py

```diff
@@ -1,63 +1,326 @@
+import plotly.express as px
 from src.digitaltwin import setup_environment
-from src.dynamic_boundary_conditions import rainfall_sites
-from src.dynamic_boundary_conditions import thiessen_polygon_calculator
-from src.dynamic_boundary_conditions import hirds_rainfall_data_to_db
-from src.dynamic_boundary_conditions import hirds_rainfall_data_from_db
+from src.dynamic_boundary_conditions import main_rainfall, thiessen_polygons, hirds_rainfall_data_from_db
 
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
 
-def catchment_area_geometry_info(catchment_file_path) -> Polygon:
+formatter = logging.Formatter("%(levelname)s:%(asctime)s:%(name)s:%(message)s")
+stream_handler = logging.StreamHandler()
+stream_handler.setFormatter(formatter)
+
+log.addHandler(stream_handler)
+
+
+def get_transposed_data(rain_depth_in_catchment: pd.DataFrame) -> pd.DataFrame:
     """
-    Extract shapely geometry polygon from the catchment file.
+    Clean and transpose the retrieved scenario data from the database for sites within the catchment area and
+    return in Pandas DataFrame format.
 
     Parameters
     ----------
-    catchment_file_path
-        The file path of the catchment polygon shapefile.
+    rain_depth_in_catchment : pd.DataFrame
+        Rainfall depths for sites within the catchment area for a specified scenario retrieved from the database.
     """
-    catchment = gpd.read_file(catchment_file_path)
-    catchment = catchment.to_crs(4326)
-    catchment_polygon = catchment["geometry"][0]
-    return catchment_polygon
+
+    # drop unnecessary columns
+    catchment_data = rain_depth_in_catchment.drop(columns=["category", "rcp", "time_period", "ari", "aep"])
+    # change column names
+    for index, column_name in enumerate(catchment_data.columns):
+        if column_name.endswith("m"):
+            catchment_data.columns.values[index] = int(column_name[:-1])
+        elif column_name.endswith("h"):
+            catchment_data.columns.values[index] = int(column_name[:-1]) * 60
+    # transpose data frame
+    transposed_catchment_data = (
+        catchment_data.set_index("site_id").rename_axis(None).transpose().rename_axis("duration_mins").reset_index()
+    )
+    return transposed_catchment_data
+
+
+def get_interpolated_data(
+        transposed_catchment_data: pd.DataFrame,
+        increment_mins: int,
+        interp_method: str) -> pd.DataFrame:
+    """
+    Perform temporal interpolation on the transposed scenario data for sites within the catchment area and
+    return in Pandas DataFrame format.
+
+    Parameters
+    ----------
+    transposed_catchment_data : pd.DataFrame
+        Transposed scenario data retrieved from the database.
+    increment_mins : int
+        Time interval in minutes.
+    interp_method : str
+        Temporal interpolation method to be used. One of 'linear', 'nearest', 'nearest-up', 'zero',
+ |      'slinear', 'quadratic', 'cubic', 'previous', or 'next'.
```

Comment:
Instead of using `str` I think you should be using either `Literal` or `Enum`. I would lean towards `Enum`

Reply:
As discussed, this does not require a change since `interp_method` is based on available methods from scipy.interpolate.interp1d() which might be subjective to changes etc. Have updated docstring mentioning scipy.interpolate.interp1d().